ui/app/mixins/with-nav-to-nearest-ancestor.js

```diff
@@ -0,0 +1,40 @@
+import Mixin from '@ember/object/mixin';
+import utils from 'vault/lib/key-utils';
+import { task } from 'ember-concurrency';
+
+// This mixin is currently used in a controller and a component, but we
+// don't see cancellation of the task as the while loop runs in either
+
```

Comment:
Ah yeah - in this case `transitionToRoute` was already a method on the component (we implemented it since we're routing there rather than bubbling out to the controller/route).

This mixin is also very specific to the secrets list route - it wouldn't be used elsewhere.